pkg/cleaner/cleaning.go

```diff
@@ -0,0 +1,138 @@
+package cleaner
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/must-gather-clean/pkg/kube"
+	"github.com/openshift/must-gather-clean/pkg/obfuscator"
+	"github.com/openshift/must-gather-clean/pkg/omitter"
+)
+
+type CleaningProcessor interface {
+	// ProcessFile is the end2end method for cleaning (omit + obfuscate path and content of) a file on disk.
+	// Will return nil if the file was processed without an error (e.g. through omission) or the error otherwise.
+	ProcessFile(inputFile string) error
+}
+
+type ReaderWriterObfuscator interface {
+	// ObfuscateInputReader obfuscates on an agnostic line-based reader and writes to an agnostic writer facility
+	ObfuscateInputReader(inputReader io.Reader, outputWriter io.Writer) error
+}
+
+type FileObfuscator interface {
+	// ObfuscatePlainTextFile obfuscates a text file and writes the result into the outputFile.
+	ObfuscatePlainTextFile(inputFile string, outputFile string) error
+}
+
+type ContentObfuscator struct {
+	Obfuscator obfuscator.ReportingObfuscator
+}
+
+type FileCleaner struct {
+	ContentObfuscator
+
+	inputFolder  string
+	outputFolder string
+
+	omitter omitter.ReportingOmitter
+}
+
+func (c *FileCleaner) ProcessFile(path string) error {
+	omit, err := c.omitter.OmitPath(path)
+	if err != nil {
+		return err
+	}
+
+	if omit {
+		return nil
+	}
+
+	isKubernetesResource := true
+	kubeResource, err := kube.ReadKubernetesResourceFromPath(filepath.Join(c.inputFolder, path))
+	if err != nil {
+		if err == kube.NoKubernetesResourceError {
+			isKubernetesResource = false
+		} else {
+			return err
+		}
+	}
+
+	if isKubernetesResource {
+		omit, err := c.omitter.OmitKubeResource(kubeResource)
+		if err != nil {
+			return err
+		}
+
+		if omit {
+			return nil
+		}
+	}
+
+	// obfuscate the text file with updated path name, which can also contain confidential information
+	return c.ObfuscatePlainTextFile(path, c.ContentObfuscator.Obfuscator.Path(path))
+}
+
+func (c *FileCleaner) ObfuscatePlainTextFile(inputFile string, outputFile string) error {
+	readPath := filepath.Join(c.inputFolder, inputFile)
+	writePath := filepath.Join(c.outputFolder, outputFile)
+	writePathParentDir := filepath.Dir(writePath)
+
+	inputOsFile, err := os.Open(readPath)
+	if err != nil {
+		return err
+	}
+
+	err = os.MkdirAll(writePathParentDir, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create directory %s: %c", writePathParentDir, err)
+	}
+
+	outputOsFile, err := os.OpenFile(writePath, os.O_CREATE|os.O_WRONLY, 0666)
```

Comment:
I assume we don't use `os.Create` here because we want it to error when the file already exists?

Reply:
you have used OpenFile here: https://github.com/openshift/must-gather-clean/blob/24c96bc6cd283ace209a1f22e45d0f692f508816/pkg/output/output.go#L39

Appending is certainly not what we want, is it?

Reply:
Yep, the current logic is not correct.